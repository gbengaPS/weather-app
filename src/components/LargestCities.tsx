import React from 'react';
import { getIconURL } from '../helpers';
import { countriesMap } from '../constants';
import FavoriteButton from './FavoriteButton';
import { Link } from 'react-router-dom';
import { WeatherInfo } from '../types';

interface LargestCitiesProps {
  cities: Array<any>,
  onRemoveItem: Function,
  favoritesMap: any,
  onFavoriteClick: Function
}

const LargestCities = (props: LargestCitiesProps) => {
  const { cities, onRemoveItem, onFavoriteClick, favoritesMap } = props;

  if (!cities) return null;
 
  return <ul className="city-list">
    {cities.map((city: WeatherInfo) => {
      const { data } = city;
      return (
        <li key={data.name} >
          <div className="city-info">
            <span>
              <Link to={`/${city}/details`}>
                {data.name}, {countriesMap[data.sys.country]}
              </Link>
            </span>
            <div className="aligned-flex">
              <span className="temperature">
                {Math.floor(data.main.temp)}&#8451;
              </span>
              <img src={getIconURL(data.weather[0].icon)} alt={data.weather[0].description} />
            </div>
          </div>
          <div className="">
            <button onClick={() => onRemoveItem(data)}>
              <i className="fa fa-trash" aria-hidden="true"></i>
            </button>
            <FavoriteButton
              city={data}
              onFavoriteClick={onFavoriteClick}
              favoritesMap={favoritesMap}
            />
          </div>
          
        </li>
      );
    })}
    </ul>
}

export default LargestCities;