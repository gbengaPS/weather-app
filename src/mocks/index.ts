export const citiesMock = {
  base: "stations",
  clouds: {all: 0},
  cod: 200,
  coord: {lon: 116.4, lat: 39.91},
  dt: 1602226045,
  id: 1816670,
  main: {temp: 23.52, feels_like: 22.22, temp_min: 23.33, temp_max: 23.89, pressure: 1014, humidity: 47},
  name: "Beijing",
  sys: {type: 3, id: 2000022, country: 'CN', sunrise: 1602195502, sunset: 1602236679},
  timezone: 28800,
  visibility: 10000,
  weather: [{id: 800, main: "Clear", description: "clear sky", icon: "01d"}],
  wind: {speed: 2.55, deg: 209},
};

export const initialStateMock =  {
  favorites: { 'Beijing': { data: citiesMock } },
  weatherInfo: [],
  notes: { 'Beijing': 'hello world' },
  isLoading: false
}