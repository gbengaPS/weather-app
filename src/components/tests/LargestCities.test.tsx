import React from 'react';
import { render } from '@testing-library/react';
import LargestCities from '../LargestCities';
import { citiesMock } from '../../mocks';


jest.mock('../CityListItem', () => jest.fn()
.mockImplementation(() => <>**CityListItem**</>))
describe('LargestCities', () => {
  const props = { cities: [{ data: citiesMock }] };

  it('should match snapshot when cities are empty', () => {
    const { container } = render(<LargestCities cities={[]}/>);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot when cities are present', () => {
    const { container } = render(<LargestCities {...props}/>);
    expect(container).toMatchSnapshot();
  });
});
