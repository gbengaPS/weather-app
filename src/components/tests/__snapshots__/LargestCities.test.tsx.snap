// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargestCities should match snapshot when cities are empty 1`] = `
<div>
  <p
    class="error"
  >
    No weather information to display.
  </p>
</div>
`;

exports[`LargestCities should match snapshot when cities are present 1`] = `
<div>
  <ul
    class="city-list"
  >
    **CityListItem**
  </ul>
</div>
`;
